tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,314 @@
+we want to organize those axes into a regular grid.  Matplotlib has a
+variety of tools for working with grids of axes that have evolved over
+the history of the library.  Here we will discuss the tools we think
+users should use most often, and then dicuss some of the older tools,
+and the tools that underpin how axes are organized.
+
+How to create grid-shaped combinations of axes:
+
+`~matplotlib.pyplot.subplots`
+    The primary function used to create figures and a grid of axes.  It is
+    similar to `.pyplot.subplot`, but creates and places all axes on the
+    figure at once, and returns an object array with handles for
+    the axes in the grid.  See also `.Figure.subplots`.
+
+or
+
+`~matplotlib.pyplot.subplot_mosaic`
+    A more flexible function used to create figures and a grid of axes,
+    with the added flexibility that some of the axes can span rows or
+    columns, and that the axes are returned in a labelled dictionary instead
+    of an array.  See also `.Figure.subplot_mosaic`.
+
+Sometimes it is natural to have more than one distinct group of axes grids,
+in which case Matplotlib has the concept of `~.figure.SubFigure`:
+
+`~matplotlib.figure.SubFigure`
+    A virtual figure within a figure.
+
+Underlying these is the concept of a `~.gridspec.GridSpec` and
+`~.SubplotSpec`:
+
+`~matplotlib.gridspec.GridSpec`
+    Specifies the geometry of the grid that a subplot will be
+    placed. The number of rows and number of columns of the grid
+    need to be set. Optionally, the subplot layout parameters
+    (e.g., left, right, etc.) can be tuned.
+
+`~matplotlib.gridspec.SubplotSpec`
+    Specifies the location of the subplot in the given `.GridSpec`.
+
+An older way to add axes in grids:
+
+`~matplotlib.pyplot.subplot2grid`
+    A helper function that is similar to `.pyplot.subplot`,
+    but uses 0-based indexing and let subplot to occupy multiple cells.
+    This function is not covered in this tutorial.
+
+.. redirect-from:: /tutorials/intermediate/gridspec
+
+"""
+
+import matplotlib.pyplot as plt
+
+import numpy as np
+
+############################################################################
+# High-level methods for making grids
+# ===================================
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can create a basic 2-by-2 grid of axes using
+# :func:`~matplotlib.pyplot.subplots`.  It returns a
+# :class:`~matplotlib.figure.Figure` instance and an array of
+# :class:`~matplotlib.axes.Axes` objects.  The axes objects can
+# be used to access methods to place artists on the axes; here we
+# use `~.Axes.annotate`, but other examples could be `~.Axes.plot`,
+# `~.Axes.pcolormesh`, etc.
+
+fig, axs = plt.subplots(ncols=2, nrows=2, figsize=(4.5, 3.5),
+                        constrained_layout=True)
+for row in range(2):
+    for col in range(2):
+        axs[row, col].annotate(f'axs[{row}, {col}]', (0.1, 0.5),
+                               xycoords='axes fraction', va='center')
```

Comment:
```suggestion
annotate_axes(axs[0, 0], 'axs[0, 0]')    
annotate_axes(axs[0, 1], 'axs[0, 1]')
annotate_axes(axs[1, 0], 'axs[1, 0]')
annotate_axes(axs[1, 1], 'axs[1, 1]')
```

and add the following at top-level below the imports:
```
def annotate_axes(ax, text):
    ax.text(0.5, 0.5, text, transform=ax.transAxes,
            ha="center", va="center", fontsize=48, color="darkgrey")
```

This is similar to the annotation method in https://matplotlib.org/devdocs/tutorials/provisional/mosaic.html#sphx-glr-tutorials-provisional-mosaic-py. IMHO this gives nicer plots. Also the call `annotate_axes(axs[0, 0], 'my text')` is quite instructive for the present purpose: It exhibits the Axes accessing method `axs[0, 0]` right next to the text, which is displayed in the figure. This should make metally connecting the figure with the code quite easy and hides all the text configuration details. 

This method should be used throughout the tutorial.

Reply:
I thought about doing that, but ultimately decided against.  I think users find the object-oriented interface confusing, so showing it _directly_ calling methods is what I was after here, using the objects they just made.  Passing the object to a function is definitely more elegant, but makes it less obvious what is going on.

Reply:
Is a double-nested loop, plus f-string labels, plus a bunch of formatting parameters really making it more obvious what the code does? Are the details even relevant?

Anyway, I do care about visual impressions and And I believe that
![grafik](https://user-images.githubusercontent.com/2836374/141868535-9f93c632-f39a-40c5-b3fd-4f7e49222a43.png)

is objectively better than
![grafik](https://user-images.githubusercontent.com/2836374/141868426-102b73ab-2e1c-4dde-8467-9c952a0df7d0.png)

(Sorry for the blurred images. Screeenshots on hi-dpi displays a nuisance.)

If we can make that work with your explicit code, that'd be great. But possibly means you need some more parameters. Maybe you could extract all the parameters to something like
```
axs[row, col].annotate(f'axs[{row}, {col}]', (0.1, 0.5), **fmt_params)
```

Reply:
OK, I've done it "normally" the first time, and then wrapped it for subsequent annotations.  That also gets across the useful way of wrapping axes-level styling functions...